### PR TITLE
First version of Movement and Wall Running

### PR DESCRIPTION
**Pull request by neoJINXD**
Currently has the movement system and a first version of the wallrunning system based on collisions.

Might show merge conflicts, but am opening up the PR earlier before fixing them
Will be done in a commit before the merge

### CODE REVIEW

**Comment by neoJINXD** [comment]
Will merge this one after the one of Ernest is merged to fixed potential conflicts with that

**Review by GAJ98 — APPROVED** [review]
Movement is all good, except for the map issue i mentioned

**Pull request merged**
